src/main/rx/rx.c

```diff
@@ -658,7 +662,15 @@ static void readRxChannelsApplyRanges(void)
         } else
 #endif
         {
+#if defined(USE_RX_MSP_OVERWRITE)
+            if (mspOverwriteState && (channel < NON_AUX_CHANNEL_COUNT)) {
+                sample = rxMspReadRawRC(&rxRuntimeState, rawChannel);
+            } else {
+                sample = rxRuntimeState.rcReadRawFn(&rxRuntimeState, rawChannel);
+            }
+#else
             sample = rxRuntimeState.rcReadRawFn(&rxRuntimeState, rawChannel);
+#endif
```

Comment:
Why’s use do while ? Preprocessor will cut out this section of code `#if defined(USE_RX_MSP_OVERWRITE) ... #endif`

Reply:
The problem is that `sample = rxRuntimeState.rcReadRawFn(&rxRuntimeState, rawChannel);` statement is duplicated, depending on USE_RX_MSP_OVERWRITE. This can break with further refactoring.

another options is to use 
```
if (false) { }
#if defined(USE_RX_MSP_OVERWRITE)
else if (mspOverwriteState && (channel < NON_AUX_CHANNEL_COUNT)) {
                sample = rxMspReadRawRC(&rxRuntimeState, rawChannel);
            }
#else
else {
    sample = rxRuntimeState.rcReadRawFn(&rxRuntimeState, rawChannel);
}

Reply:
:thinking: as the code is isolated we should not add it by default. This should be a custom define in the [cloud] build.

@ledvinap Does this not result in endless loop if [USE_RX_MSP_OVERWRITE](url) were not defined?

```
            do {
#if defined(USE_RX_MSP_OVERWRITE)
                if (mspOverwriteState && (channel < NON_AUX_CHANNEL_COUNT)) {
                    sample = rxMspReadRawRC(&rxRuntimeState, rawChannel);
                    break;
                }
#endif
                sample = rxRuntimeState.rcReadRawFn(&rxRuntimeState, rawChannel);
            } while (0);
```

@cvetaevvitaliy Also for testing - the box mode does not show up in configurator - as I am busy with the 4.4.2 release and support - could you provide details (for documentation)?

Reply:
@haslinghuis : The block will run only once, loop condition is always false - `while (0);`. But using `if (false) {} else if (test) {} else {}` as in other comment is probably cleaner here.

Reply:
Thanks, very obvious now on a new day.